wgpu-types/src/lib.rs

```diff
@@ -1304,7 +1338,133 @@ impl TextureFormat {
             block_dimensions,
             block_size,
             srgb,
+        }
+    }
+
+    /// Lists features supported by WebGPU, not using any native extensions.
+    pub fn features_webgpu(&self) -> TextureFormatFeatures {
```

Comment:
why would we want to move it out into a separate function? we could still have it all in the `TextureFormatInfo` and have a single `match` over the formats.

Reply:
TextureFormatInfo describes static information about each format. The "true" feature set of a texture format is now determined at runtime (unless WebGPU). Having a "features iff you don't enable the beyond-webgpu-spec-thing"-field felt a bit clumsy and confusing to me - a pure non-web application may never be interested in the value and it could confuse users.

No strong feelings about it though, so if you dislike the second gargantum-match I can fuse them together, just gotta find a good name for the TextureFormatFeatures field 🤷

Reply:
we can have this static information be more explicit, i.e. `guaranteed_features` instead of `features` as the field name would nail it